### PR TITLE
Add classifications to NOAA GUI

### PR DESCRIPTION
**Pull request by mwoehlke-kitware**
Improve our general infrastructure to support object type classifications, which, per KWIVER's design, are supported at a per-detection level. (For tracks, right now we just return the classification of the last state.) Modify the NOAA GUI to include columns for best type/score.

Reading and writing classifications is supported. Editing thereof is not yet implemented.

### CODE REVIEW

File: sealtk/core/KwiverDetectionsSink.cpp [hunk 1]
```diff
@@ -152,10 +153,12 @@ void KwiverDetectionsSink::writeData(QUrl const& uri) const
       for (auto const& i : f.detections)
       {
         auto const& qbox = d->model->data(i, AreaLocationRole).toRectF();
+        auto const& type = d->model->data(i, ClassificationRole).toHash();
         auto kbox = kv::bounding_box_d{{qbox.left(), qbox.top()},
                                        qbox.width(), qbox.height()};
```
Comment (KyleFromKitware) [inline sealtk/core/KwiverDetectionsSink.cpp]: This pattern has come up often enough that we may want to eventually make a dedicated `QRectF` <-> `bounding_box_d` converter class, but that is beyond the scope of this PR.

Comment (mwoehlke-kitware) [reply sealtk/core/KwiverDetectionsSink.cpp]: You say that, but I *did* add `TrackUtils` in this PR. OTOH I only see two instances of this so far, and the inline code isn't unreasonable (unlike `classificationToDetectedObjectType`, which can't practically be inlined, and which is going to have a third user when type *editing* gets PR'd).

File: sealtk/core/test/TestTracks.cpp [hunk 2]
```diff
@@ -29,15 +29,26 @@ namespace test
...
   QVERIFY(model.data(index, AreaLocationRole).canConvert<QRectF>());
-  QCOMPARE(model.data(index, AreaLocationRole).value<QRectF>(), box);
+  QCOMPARE(model.data(index, AreaLocationRole).toRectF(),
+           state.location);
+
+  auto const modelHasClassification =
+    !model.data(index, ClassificationRole).isValid();
+  if (state.classification.isEmpty() == !modelHasClassification)
+  {
+    qDebug() << state.classification << model.data(index, ClassificationRole);
```
Comment (KyleFromKitware) [inline sealtk/core/test/TestTracks.cpp]: Please remove the stray debug statement.

File: sealtk/core/test/TestTracks.cpp [hunk 3]
```diff
@@ -29,15 +29,26 @@ namespace test
...
+                   QModelIndex const& index, TrackState const& state)
 {
   QVERIFY(model.data(index, AreaLocationRole).canConvert<QRectF>());
-  QCOMPARE(model.data(index, AreaLocationRole).value<QRectF>(), box);
+  QCOMPARE(model.data(index, AreaLocationRole).toRectF(),
+           state.location);
+
+  auto const modelHasClassification =
+    !model.data(index, ClassificationRole).isValid();
+  if (state.classification.isEmpty() == !modelHasClassification)
```
Comment (KyleFromKitware) [inline sealtk/core/test/TestTracks.cpp]: Shouldn't we be asserting that these two are equal, and then doing the contents of the if-block if `modelHasClassification` is true?

Comment (mwoehlke-kitware) [reply sealtk/core/test/TestTracks.cpp]: Uh... something like that. Please see if the updated version is better.

(If there is no classification, both `QVariant{}` and `QVariant{QVariantHash{}}` are acceptable results.)

File: sealtk/core/test/TestTracks.cpp [hunk 4]
```diff
@@ -37,9 +37,12 @@ void testTrackData(QAbstractItemModel const& model,
...
   auto const modelHasClassification =
     !model.data(index, ClassificationRole).isValid();
-  if (state.classification.isEmpty() == !modelHasClassification)
+  if (!modelHasClassification)
+  {
+    QVERIFY(state.classification.isEmpty());
+  }
+  else
   {
-    qDebug() << state.classification << model.data(index, ClassificationRole);
```
Comment (KyleFromKitware) [inline sealtk/core/test/TestTracks.cpp]: This removal was not in the same commit that introduced it in the first place. I'm OK with not having a "pure" history, so it's up to you if you want to fix it.

Comment (mwoehlke-kitware) [reply sealtk/core/test/TestTracks.cpp]: Ack, no, I'll fix it... possibly I squashed the fixup commit into the wrong spot :cry:.

Comment (mwoehlke-kitware) [reply sealtk/core/test/TestTracks.cpp]: Fixed; thank you for noticing!!!

File: sealtk/core/test/TestTracks.cpp [hunk 5]
```diff
@@ -37,7 +37,11 @@ void testTrackData(QAbstractItemModel const& model,
 
   auto const modelHasClassification =
     !model.data(index, ClassificationRole).isValid();
-  if (state.classification.isEmpty() == !modelHasClassification)
+  if (!modelHasClassification)
+  {
+    QVERIFY(state.classification.isEmpty());
+  }
+  else
```
Comment (KyleFromKitware) [inline sealtk/core/test/TestTracks.cpp]: This fix was *also* not in the correct commit.

Comment (KyleFromKitware) [reply sealtk/core/test/TestTracks.cpp]: But I think the fix itself does look correct.

**Comment by KyleFromKitware** [comment]
The `KwiverTrackModel:data` and `KwiverTrackSource:loadTracks` tests fail in local testing.

**Pull request merged**